extensions/amp-access/0.1/amp-access.js

```diff
@@ -770,17 +777,43 @@ export class AccessService {
   }
 
   /**
+   * Runs the login flow using one of the predefined urls in the amp-access config
+   *
+   * @param {string} type Type of login defined in the config
+   * @return {!Promise}
+   */
+  loginWithType_(type) {
+    // Login URL should always be available at this time.
+    const loginUrl = user().assert(this.loginUrlMap_[type],
+        'Login URL is not ready: %s', type);
+    user().assert(this.loginConfig_[type],
+        'Login URL is not configured: %s', type);
+    return this.login_(loginUrl, type);
+  }
+
+  /**
+   * Runs the login flow opening the given url in the login window.
+   *
+   * @param {string} url
+   * @return {!Promise}
+   */
+  loginWithUrl(url) {
+    return this.login_(url, url);
```

Comment:
We can't use `url` as an analytics label. Two options we have:

1. No label at all (eventLabel == empty string). This will result in analytics events that look like this: `access-login-success`.
2. A label by the name of you source, e.g. "laterpay". The analytics event would look like `access-login-laterpay-success`. You can pass "laterpay" as an argument to this method.

Overall, I'd say, if you only supply "laterpay" as an event type, there's no a lot of point in having it at all and we can just go with `''`. However, of you'll have a couple of different login types, it might be a good idea to provide them. E.g. "laterpay-login1", "laterpay-login2" - this sort of thing.